### PR TITLE
Make the logic for checking scatter vectorized nodes of GEP clearer

### PR DESCRIPTION
**Pull request by tcwzxx**
There is no functional change.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -6820,9 +6821,7 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
 
   // Special processing for sorted pointers for ScatterVectorize node with
   // constant indeces only.
-  if (AreAllSameInsts && UserTreeIdx.UserTE &&
-      UserTreeIdx.UserTE->State == TreeEntry::ScatterVectorize &&
-      !(S.getOpcode() && allSameBlock(VL))) {
+  if (not AreAllSameBlock && AreScatterAllGEPSameBlock) {
```
Comment (tschuett) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:6824]: `not` is not used in the repo. Please use `!`.

**Comment by tschuett** [comment]
I missed the file name.

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp (hunk 1 above)
Comment (tcwzxx) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp]: Thanks, fixed.

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 2]
```diff
@@ -6725,9 +6725,9 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
   bool IsScatterVectorizeUserTE =
       UserTreeIdx.UserTE &&
       UserTreeIdx.UserTE->State == TreeEntry::ScatterVectorize;
-  bool AreAllSameInsts =
-      (S.getOpcode() && allSameBlock(VL)) ||
-      (S.OpValue->getType()->isPointerTy() && IsScatterVectorizeUserTE &&
+  bool AreAllSameBlock = (S.getOpcode() && allSameBlock(VL));
```
Comment (alexey-bataev) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:6728]: Remove extra parens

**Review by alexey-bataev — APPROVED** [review]
LG with a nit

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp (hunk 2 above)
Comment (tcwzxx) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp]: Thanks, fixed.

Comment (tcwzxx) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp]: Since I don't have write permissions, could you help me merge?

**Pull request merged**